### ISSUE TITLE
Why not support Complex instance field value?

### ISSUE THREAD

**Issue by nile** [st-js/st-js#112]
I want to create some code like this

    var UsersRouter = React.createClass({
        mixins: [RouterMixin],

        routes: {
            '/': 'users',
            '/create': 'create',
            '/view/:id': 'view'
         },
      ....
    }

What could I do?

**Comment by onigoetz** [comment]
I don't understand, what is the part you can't do ?

With the react bridge, the "React.createClass" is done automatically.

For your example of complex fields, the only way is to add them as "static" in the class or else you can't initialize the field

Does that answer your question ?

**Comment by nile** [comment]
Thank you for your work, stjs is great.
I use the react bridge and use static to initliaze the complex fields, but that cause React error.

https://github.com/onigoetz/react-bridge/issues/1

The solution isonigoetz copy 'static' to instance, that looks unnormal.

My question is, why cannot support complex instance fields in st-js?

**Comment by acraciun** [comment]
the reason is linked to how classes are usually emulated in JavaScript -
use the prototype mechanism. So a Java code like this:
public MyClass {
  private SomeType instanceField = new SomeType();
}
would be translated into something like:

var MyClass = function (){};
MyClass.prototype.instanceField = new SomeType();

The big difference between the two codes is that in Java, the field
instantiation is called when you create a new object of type MyClass, but
in the corresponding JavaScript code this will be called at class
definition - so all the instances of MyClass would initially share exactly
the same instance of SomeType! - very dangerous.
So we blocked the generation of code where the field is assigned with
mutable values (that's why we allow only primitives). The users should move
the intializations within the constructors. We could've done it
automatically, but as most of the time we tried to stay away from diverging
from the initial Java code.

  The trouble with the way the React bridge was proposed is that the
constructor of the class is not called before the React class is defined
... so that's why the static fields were used - but I agree the solution
does not look great.
